### PR TITLE
GH-94857: fix test_io refleak

### PR DESCRIPTION
**Pull request by kumaraditya303**
* Issue: gh-94857

Automerge-Triggered-By: GH:tiran

### CODE REVIEW

**Review by tiran — CHANGES_REQUESTED** [review]
There are more possible refleaks in error paths. It would be better to incref ``encodings`` eagerly and ``decref`` it in error paths.

**Review by serhiy-storchaka — COMMENTED** [review]
There are still leaks in three `return` statements between lines 1260 and 1280.

**Comment by kumaraditya303** [comment]
I have made the requested changes; please review again

File: Modules/_io/textio.c [hunk 1]
```diff
@@ -1247,6 +1247,7 @@ textiowrapper_change_encoding(textio *self, PyObject *encoding,
         if (errors == Py_None) {
             errors = self->errors;
         }
+        Py_INCREF(encoding);
     }
     else {
         if (_PyUnicode_EqualToASCIIString(encoding, "locale")) {
```
Comment (tiran) [inline Modules/_io/textio.c:1253]: You have to ``Py_INCREF(encoding)`` when ``encoding != "locale"``.

Comment (kumaraditya303) [reply Modules/_io/textio.c:1253]: Good catch! done

**Comment by kumaraditya303** [comment]
PTAL @tiran Thanks!

**Review by tiran — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**